File: daily/2025/1422_Maximum_Score_After_Splitting_a_String.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    int maxScore(string s) {
        int n = s.length();
        int left = (s[0] == '0');
        int right = 0;
        for (int i = 1; i < n; i++) right += (s[i] == '1');
        int ans = left + right;
        for (int i = 1; i < n-1; i++){
            left += (s[i] == '0');
            right -= (s[i] == '1');
            ans = max(ans, left + right);
        }
        return ans;
    }
};

int main(){
    string s = "011101";
    // string s = "00111";
    // string s = "1111";
    Solution S;
    cout << S.maxScore(s) << endl;
    return 0;
}

File: daily_python/1422_Maximum_Score_After_Splitting_a_String.py
import os
import sys
from typing import List

class Solution:
    def maxScore(self, s: str) -> int:
        left = 0 if s[0] == '1' else 1
        right = s.count('1') - (1 if s[0] == '1' else 0)
        ans = left + right
        for i in range(1, len(s)-1):
            if s[i] == '0':
                left += 1
            else:
                right -= 1
            ans = max(ans, left + right)
        return ans
    
def readDataSet(filename):
    dataset = []
    with open(filename, 'r') as file:
        content = file.read().strip()
        blocks = content.split('\n\n')
        for block in blocks:
            lines = block.split('\n')
            s = lines[0].split('=')[1].strip()[1:-2]
            dataset.append(s)
    return dataset

if __name__ == '__main__':
    if len(sys.argv) == 1:
        filename = os.path.splitext(__file__)[0] + '.txt'
    else:
        filename = sys.argv[1]
    dataset = readDataSet(filename)
    results = []
    solution = Solution()
    for s in dataset:
        results.append(solution.maxScore(s))
    for index, result in enumerate(results):
        print(f'Example {index + 1} : {result}')